primitives/src/secondary_key.rs

```diff
@@ -57,14 +64,37 @@ impl PalletPermissions {
     pub fn entire_pallet(pallet_name: PalletName) -> Self {
         PalletPermissions {
             pallet_name,
-            dispatchable_names: SubsetRestriction(None),
+            dispatchable_names: SubsetRestriction::Whole,
         }
     }
 }
 
 /// Extrinsic permissions.
 pub type ExtrinsicPermissions = SubsetRestriction<PalletPermissions>;
 
+impl ExtrinsicPermissions {
+    /// Returns `true` iff this permission set permits calling `pallet::dispatchable`.
+    pub fn sufficient_for(&self, pallet: &PalletName, dispatchable: &DispatchableName) -> bool {
+        let matches_any = |perms: &BTreeSet<PalletPermissions>| {
+            perms.iter().any(|perm| {
+                if &perm.pallet_name != pallet {
+                    return false;
+                }
+                match &perm.dispatchable_names {
+                    SubsetRestriction::Whole => true,
+                    SubsetRestriction::These(funcs) => funcs.contains(dispatchable),
+                    SubsetRestriction::Except(funcs) => !funcs.contains(dispatchable),
+                }
+            })
+        };
```

Comment:
We need to introduce a concept of complexity here and limit the complexity. It's not specific to EA but in general, there needs to be a complexity limit on extrinsic perms. Can be done in a different PR and Ticket.

Reply:
This is already enforced via `ensure_perms_length_limited` and friends.